### ISSUE TITLE
linter: `eslint(curly)` false positive when using `multi-line` setting

### ISSUE THREAD

**Issue by bfaulk96** [oxc-project/oxc#13352]
### What version of Oxlint are you using?

latest

### What command did you run?

`npx oxlint --type-aware src`

### What does your `.oxlintrc.json` config file look like?

```jsonc
{
  "plugins": [
    "typescript",
    "unicorn",
    "react",
    "oxc",
    "import",
    "node"
  ],
  "ignorePatterns": ["lib/**", "node_modules/**"],
  "globals": {
    "AWS_ENV": "readonly"
  },
  "env": {
    "es2021": true,
    "node": true,
    "commonjs": true
  },
  "rules": {
    "no-console": "error",
    "no-var": "warn",
    "no-prototype-builtins": "error",
    "no-useless-constructor": "error",
    "exhaustive-deps": "warn",
    "default-case-last": "error",
    "curly": ["error", "multi-line"],
    "no-array-constructor": "error",
    "no-case-declarations": "error",
    "no-unused-expressions": "warn",
    "no-require-imports": "warn",
    "no-extend-native": "error",
    "no-extra-bind": "error",
    "no-fallthrough": "error",
    "no-iterator": "error",
    "no-lone-blocks": "error",
    "no-multi-str": "error",
    "no-labels": [
      "error",
      {"allowLoop": false, "allowSwitch": false}
    ],
    "new-cap": [
      "error",
      {
        "newIsCap": true,
        "capIsNew": false,
        "properties": true
      }
    ],
    "ban-ts-comment": "error",
    "unbound-method": "off",
    "typescript/no-redundant-type-constituents": "off",
    "eqeqeq": "error",
    "no-empty": ["error", {"allowEmptyCatch": true}],
    "no-unused-vars": [
      "error",
       {
         "args": "after-used",
         "argsIgnorePattern": "^_",
         "ignoreRestSiblings": true
       }
    ],
    "array-callback-return": [
      "error",
      {
        "allowImplicit": false,
        "checkForEach": false,
        "allowVoid": false
      }
    ],
    "no-unknown-property": ["error", { "ignore": ["sx"] }],
    "sort-imports": ["warn", { "ignoreDeclarationSort": true }]
  }
}
```

### What happened?

Specifically with the rule `"curly": ["error", "multi-line"],`, oxlint seems to have a false positive on single-line `else` statements when the `if` statement above has an in-line comment. This behavior differs from eslint, which correctly sees both situations as "single-line" usage and therefore doesn't show the error.

<img width="1354" height="178" alt="Image" src="https://github.com/user-attachments/assets/9b48e541-ddbf-441d-8a7a-3bce25ad35df" />

<img width="1906" height="398" alt="Image" src="https://github.com/user-attachments/assets/e532336c-86cb-402c-890f-e0431b34514d" />

**Comment by telamon** [comment]
This bug is limited to one-liner conditionals that end with a comment; removing comment removes the false positive. (not gonna do that though)

oxlintrc.json
```json
{
  "rules": {
    "curly": [ "error", "multi-line" ]
  }
}
```

repro.js
```js
const bar = 3
let foo = -1

if (bar < 2) foo = 1 // comment 1
else if (bar > 3) foo = 2 // comment 2 + false positive
else foo = 3 // comment 3 + false positive

console.log(foo)
```

lint output
```bash
oxlint -c standard-oxlintrc.json -f unix repro.js 
repro.js:6:6: Expected { after 'else'. [Error/eslint(curly)]
repro.js:6:6: Expected { after 'else'. [Error/eslint(curly)]
```

**Comment by connorshea** [comment]
This is fixed now based on testing the reproduction case provided.
